src/Symfony/Component/HttpFoundation/InputBag.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation;
+
+use Symfony\Component\HttpFoundation\Exception\BadRequestException;
+
+/**
+ * InputBag is a container for user input values such as $_GET, $_POST, $_REQUEST, and $_COOKIE.
+ *
+ * @author Saif Eddin Gmati <saif.gmati@symfony.com>
+ */
+final class InputBag extends ParameterBag
+{
+    /**
+     * Returns a string input value by name.
+     *
+     * @param string|null $default The default value if the input key does not exist
+     *
+     * @return string|array|null
+     */
+    public function get(string $key, $default = null)
+    {
+        if (null !== $default && !\is_string($default)) {
```

Comment:
all `is_string()` checks in the PR are too strict, we want to accept any stringable:
`is_scalar($v) || is_callable([$v, '__toString'])`